sklearn/metrics/cluster/_supervised.py

```diff
@@ -138,6 +146,206 @@ def contingency_matrix(labels_true, labels_pred, *, eps=None, sparse=False):
+    D10 = 2 * (sum_comb_c - sum_comb)
+    D00 = n_samples * n_samples - n_samples - D01 - D10 - D11
+    return np.array([[D00, D01], [D10, D11]])
+
+
+def rand_score(labels_true, labels_pred):
+    """Rand index.
+
+    The Rand Index computes a similarity measure between two clusterings
+    by considering all pairs of samples and counting pairs that are
+    assigned in the same or different clusters in the predicted and
+    true clusterings.
+
+    The raw RI score is:
+
+        RI = (D00 + D11) / (number of pairs)
+
+    where D00 = number of pairs with both clusterings having the samples
+    not clustered together, and D11 = number of pairs with both clusterings
+    having the samples clustered together.
+
+    RI is a symmetric measure::
+
+        rand_score(a, b) == rand_score(b, a)
+
+    Parameters
+    ----------
+    labels_true : int array, shape = [n_samples]
+        Ground truth class labels to be used as a reference
+
+    labels_pred : array, shape = [n_samples]
+        Cluster labels to evaluate
+
+    Returns
+    -------
+    ri : float
+       Similarity score between 0.0 and 1.0, inclusive, 1.0 stands for
+       perfect match
+
+    Examples
+    --------
+    Perfectly matching labelings have a score of 1 even
+
+      >>> from sklearn.metrics.cluster import rand_score
+      >>> rand_score([0, 0, 1, 1], [0, 0, 1, 1])
+      1.0
+      >>> rand_score([0, 0, 1, 1], [1, 1, 0, 0])
+      1.0
+
+    Labelings that assign all classes members to the same clusters
+    are complete but may be not always pure, hence penalized:
+
+      >>> rand_score([0, 0, 1, 2], [0, 0, 1, 1])
+      0.83...
+
+    RI is symmetric, so labelings that have pure clusters with members
+    coming from the same classes but unnecessary splits are penalized::
+
+      >>> rand_score([0, 0, 1, 1], [0, 0, 1, 2])
+      0.83...
+
+    If classes members are completely split across different clusters, the
+    assignment is totally incomplete, hence the RI is minimized:
+
+      >>> rand_score([0, 0, 0, 0], [0, 1, 2, 3])
+      0.0
+
+    References
+    ----------
+    .. L. Hubert and P. Arabie, Comparing Partitions, Journal of
+      Classification 1985
+      https://link.springer.com/article/10.1007%2FBF01908075
+
+    .. https://en.wikipedia.org/wiki/Simple_matching_coefficient
+
+    .. https://en.wikipedia.org/wiki/Rand_index
+
+    See also
+    --------
+    adjusted_rand_score: Adjusted Rand Score
+    adjusted_mutual_info_score: Adjusted Mutual Information
+
+    """
+    m = pair_confusion_matrix(labels_true, labels_pred)
+    m_diag = m[0][0] + m[1][1]
+    m_sum = m_diag + m[0][1] + m[1][0]
+
+    # Special limit cases: no clustering since the data is not split;
+    # or trivial clustering where each document is assigned a unique cluster.
+    # These are perfect matches hence return 1.0.
+    if m_diag == m_sum or m_sum == 0:
+        return 1.0
+
+    return m_diag / m_sum
+
+
 def adjusted_rand_score(labels_true, labels_pred):
```

Comment:
Shouldn't we use `pair_confusion_matrix` in `adjusted_rand_score` too?

Reply:
I would leave it as is. The code currently reflects how the main wikipedia page on the rand index is written [https://en.wikipedia.org/wiki/Rand_index]. Both wikipedia and scikit have for the raw rand index the computation via the pair confusion matrix and for the adjusted rand score via the contingency table.